### ISSUE TITLE
Strings documentation regarding backslash

### ISSUE THREAD

**Issue by greatgraphicdesign** [php/doc-en#1804]
ref: https://www.php.net/manual/en/language.types.string.php

The description under “Single quoted” says, “To specify a literal backslash, double it (\\);” however, in the example we see...
```php
// Outputs: You deleted C:\*.*?
echo 'You deleted C:\*.*?';
```
... so it seems it is not necessary to double the backslash. Should the instruction for doubling a literal backslash be removed from this section?

In the “Double quoted” sectoin we see, “As in single quoted strings, escaping any other character will result in the backslash being printed too.” This statement references a table of escaped characters that applies to double quoted strings, but not single quoted strings. I would cut the phrase “As in single quoted strings...” to avoid leading people to think that the table has anything to do with single quoted strings. It's clear enough to say, “Escaping any other character will result in the backslash being printed.”

**Comment by cmb69** [comment]
> To specify a literal single quote, escape it with a backslash (\). To specify a literal backslash, double it (\\). All other instances of backslash will be treated as a literal backslash:

IMO, that is relatively clear: you don't *need* to double the backslash (unless it is followed by `'`), but doubling is always possible, and usually a good idea (unless in regex patterns).

**Comment by greatgraphicdesign** [comment]
Is this more clear?

> To specify a literal single quote, escape it with a backslash (\\'). A single backslash (\\) is treated as a literal backslash, but you can also specify a literal backslash by doubling it (\\\\). All other instances of backslash will be treated as a literal backslash:

**Comment by cmb69** [comment]
Yeah, that might be better (although we should change the grammar to avoid personalization).

What do others think about this?

**Comment by Girgias** [comment]
```
To specify a literal single quote, it needs to be escaped with a backslash (\').
In all other cases, a single backslash (\) is treated as a literal backslash, however it is also possible to specify a literal backslash by escaping it (\\).
Therefore, other commonly used escape sequences such as \r or \n will be output literally as specified rather than having any special meaning. 
```

Sounds better to me as it gets rid of the personalisation and the weird fact that we're explaining that a backslash is interpreted literally while just after saying it again.

**Comment by greatgraphicdesign** [comment]
Also, when a request string (e.g., $_POST['something']) is assigned to a variable — every character is passed literally. This might be worth mentioning somewhere. Thanks for considering this feedback. :)

**Comment by hormus** [comment]
A created string literal must be escaped from the "single quote" or "double quote" delimiters as the language interprets it as leading and trailing, the "backslash" character must be escaped however php interprets correctly even if it is not doubled. There are exceptions in a regex that needs to be doubled further if its output is to appear as the "backslash" is a special character, the value other than the delimiters, e.g. file string and/or the value of a variable is already interpreted but for security reasons the backtick character must be escaped like shell_exec() etc.

**Comment by hormus** [comment]
A created string literal must be escaped from the "single quote" or "double quote" delimiters as the language interprets it as leading and trailing, the "backslash" character must be escaped however php interprets correctly even if it is not doubled. There are exceptions in a regex that needs to be doubled further if its output is to appear as the "backslash" is a special character, the value other than the delimiters, e.g. file string and/or the value of a variable is already interpreted but for security reasons the backtick character must be escaped like shell_exec() etc.
